tests/on-device/config-class.bats

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2023, see AUTHORS. Licensed under MIT License, see LICENSE.
+
+load lib
+
+@test 'successfully loads a config with _class="nixOnDroid"' {
+  # set up / build / activate the configuration
+  echo '{ config.system.stateVersion = "24.05"; _class = "nixOnDroid"; }' > ~/.config/nixpkgs/nix-on-droid.nix
+  _sed -e "s|<<FLAKE_URL>>|$FLAKE_URL|g" -e "s|<<SYSTEM>>|$(detect_system)|g" \
+    "$ON_DEVICE_TESTS_DIR/config-flake.nix" \
+    > ~/.config/nixpkgs/flake.nix
+
+  nix-on-droid switch --flake ~/.config/nixpkgs#device
+}
+
+@test 'fails to load a config with _class="nixos"' {
+  # set up / build / activate the configuration
+  echo '{ config.system.stateVersion = "24.05"; _class = "nixos"; }' > ~/.config/nixpkgs/nix-on-droid.nix
```

Comment:
I was thinking of less synthetic examples of somehow (accidentally?) trying to use nixos modules, but I'm not sure how cumbersome that'd be to write.

Reply:
ok then, but then it's not that useful yet, I guess